ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/JsonRpcParameter.java

```diff
@@ -60,11 +57,9 @@ public <T> T required(final Object[] params, final int index, final Class<T> par
    * @param <T> The type of parameter.
    * @return Returns the parameter cast as T if available.
    */
-  // TODO: update to throw JsonRpcParameterException as a checked exception, forcing callers to
-  // handle it to supply appropriate context
   @SuppressWarnings("unchecked")
```

Comment:
should this suppression be removed now?

Reply:
This was unrelated to the changes I made. It seems we're retrieving an Object from the request context and performing an unchecked cast to a parameterized type. i.e.
`
final T param;
    final Object rawParam = params[index];
    if (paramClass.isAssignableFrom(rawParam.getClass())) {
      // If we're dealing with a simple type, just cast the value
      param = (T) rawParam;
    } 
`
Looks like even though we're checking the type of T is appropriate for the object, the compiler (or intelliJ) is still complaining about the unchecked cast. We can get around this by using `paramClass.cast(rawParam)`, which should have the same effect.